### PR TITLE
scripts: add build-matrix script to ease matching ci

### PR DESCRIPTION
**Pull request by trws**
problem: we have good scripts for running in our images locally, but not always with all the same flags as in CI

solution: add a script, build-matrix, that takes a name from the actions build matrix and runs docker_run_checks with the same flags as that matrix entry

### CODE REVIEW

File: scripts/build-matrix [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+p = argparse.ArgumentParser()
+p.add_argument("-p","--platform", default=platform.machine())
+p.add_argument("--interactive", action="store_true")
+p.add_argument("name")
+args = p.parse_args()
+
+my_path = Path(__file__).absolute()
+generate_matrix = my_path.parent.parent.joinpath("src/test/generate-matrix.py")
+docker_run_checks = my_path.parent.parent.joinpath("src/test/docker/docker-run-checks.sh")
```
Comment (jameshcorbett) [inline scripts/build-matrix:22]: unused variable

File: scripts/build-matrix [hunk 2]
```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+import argparse
+from pprint import pprint
+import json
+import os
+from pathlib import Path
+from re import L
```
Comment (jameshcorbett) [inline scripts/build-matrix:8]: unused variable

**Review by jameshcorbett — APPROVED** [review]
👍

**Review by grondo — APPROVED** [review]
Great idea! I like that the script can also list the entries in the matrix.

**Comment by trws** [comment]
Thanks both of you, cleaned up a little bit and added it to the list to be processed by pre-commit so the style checks run on it.

**Pull request merged**